src/particles/particle_accretion.hpp

```diff
@@ -41,131 +36,279 @@ AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE auto compute_rho_sink(const amrex::Arra
 	return 0.2 * C::m_p;
 }
 
+template <typename problem_t>
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE auto compute_Mdot_and_r_K(const amrex::Array4<const amrex::Real> &local_state, int ix, int iy, int iz, double par_mass,
+								   double dx_max, ParticleUtils::kernel_weights_array_t kernel_weights_normalized)
+    -> std::tuple<double, double>
+{
+	// compute the average density, momentum, and sound speed in the accretion zone
+	double rho_infty = 0.0;
+	double px_infty = 0.0;
+	double py_infty = 0.0;
+	double pz_infty = 0.0;
+	double cs_infty = 0.0;
+	for (int ii = ix - stencil_size; ii <= ix + stencil_size; ++ii) {
+		for (int jj = iy - stencil_size; jj <= iy + stencil_size; ++jj) {
+			for (int kk = iz - stencil_size; kk <= iz + stencil_size; ++kk) {
+				const double weight = kernel_weights_normalized[std::abs(ii - ix)][std::abs(jj - iy)][std::abs(kk - iz)];
```

Comment:
It looks here like you're computing the weight from the difference in cell center positions, not from the position of the sink particle. I would prefer the latter, and that is what is implemented in the original 2004 paper.